### DISCUSSION TITLE
How to save added user or other things when remove container?

### DISCUSSION THREAD

**Discussion by HaoLiuHust** [minio/minio#13224] [Q&A] [answered]
I am using minio gateway to connect to a ceph cluster, after start, I added a user in minio console web, it works. Then I stop and remove the container, and start a new one, the user I added before can not work. I have set -v xxx:/root/.minio when start container

**Comment by harshavardhana** [comment, marked as answer]
Do not know what is going here @HaoLiuHust - how did you start the container? to make sure gateway keeps the same credentials all you need to set is 

```
MINIO_ROOT_USER
MINIO_ROOT_PASSWORD
```

**Reply by HaoLiuHust** [reply]
I have wrote a start_minio.sh like below:
```
sudo docker stop minio_ceph
sudo docker rm minio_ceph

sudo docker run -p 9002:9000 -p 9003:9001 --name minio_ceph \
    -d \
    -e "MINIO_ACCESS_KEY=XXX" \
    -e "MINIO_SECRET_KEY=XXX" \
    -v /home/XXX/minio_configs/config:/root/.minio \
    minio/minio \
    gateway s3 http://ceph_ip:7480 --console-address ":9001"
```

**Reply by HaoLiuHust** [reply]
and after I start up the minio, I added a user in console web
![image](https://user-images.githubusercontent.com/7233091/133548118-be854c2c-e199-4e80-a7e2-0a88e3a61453.png)

and I can login console with this user. then rerun start_minio.sh, I can not login with this user

**Reply by harshavardhana** [reply]
Of course you cannot - gateway doesn't store users locally anywhere - if you want persistence of users with Gateway you need to configure etcd service for gateway to talk to this etcd service. 

https://github.com/minio/minio/blob/master/docs/config/README.md#etcd

**Reply by HaoLiuHust** [reply]
Thanks, helps a lot, I will try it
